doc/architecture/architecture.rst

```diff
@@ -1,24 +1,234 @@
 .. _architecture:
 
 Ostro |trade| OS Architecture
-##################################
+#############################
 
-The Ostro OS is a Linux*-based operating system tailored for IoT devices and built with
-security in mind.  Ostro OS is based on standard libraries and services with 
-frameworks to help with best practices and common IoT use cases.
+Introduction
+============
 
-The following documents start off our descriptions of the Ostro |trade| OS 
-components and their relationships so developers
-can learn more technical details about what we're building.
+The Ostro |trade| OS is a pre-compiled, configured and secured base
+Internet of Things (IoT) Linux\* OS that supports creating custom images
+easily. For more information, see :ref:`about_ostro`.
 
+This document gives an overview of the Ostro OS architecture and its main
+building blocks. Furthermore, the document briefly talks about how
+Ostro OS meets different end-user use-cases ranging from hacking to
+IoT product creation.
+
+.. _`Yocto Project`: http://yoctoproject.org
+.. _`Soletta Project`: http://solettaproject.org
+.. _`Node.JS`: https://nodejs.org
+.. _`IoTivity`: https://www.iotivity.org/
+.. _`Ostro Project download server`: https://download.ostroproject.org
+.. _`application sample recipes`: https://github.com/ostroproject/meta-appfw/tree/master/recipes-appfw
+.. _`Open Connectivity Foundation`: http://openconnectivity.org/
+.. _`systemd.generator`: https://www.freedesktop.org/software/systemd/man/systemd.generator.html
+
+Ostro OS Characteristics
+========================
+
+In Ostro OS, there are four key characteristics that
+drive the architecture design:
+
+#. **Ostro OS uses standard libraries and services**: The building blocks
+   used in Ostro OS are commonly used in many open-source projects and custom
+   implementations are avoided to the extent possible. This has a big benefit
+   as there is little porting or platform adaptation needed to get
+   existing applications or services to run on Ostro OS; often times, all that's
+   needed is to create a manifest for your application.
+
+#. **Ostro OS is pre-compiled and pre-configured**: Ostro OS leverages the
+   `Yocto Project`_ build system tools and is provided as a pre-compiled and pre-configured
+   distribution specifically tuned for IoT devices. Generating and compiling your own
```

Comment:
... specifically tuned for IoT device development.

[Do we need to lightly mention here that this pre-configured stuff is for development only?]